rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/NodeSourceCreationWindow.java

```diff
@@ -0,0 +1,125 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import com.smartgwt.client.widgets.Label;
+import com.smartgwt.client.widgets.form.DynamicForm;
+import com.smartgwt.client.widgets.form.fields.CheckboxItem;
+import com.smartgwt.client.widgets.form.fields.FormItem;
+import com.smartgwt.client.widgets.form.fields.SpacerItem;
+import com.smartgwt.client.widgets.form.fields.TextItem;
+
+
+/**
+ * Dialog window to create a node source. Downloads dynamically supported
+ * infrastructures and policies when the window is created.
+ */
+public class NodeSourceCreationWindow extends NodeSourceWindow {
+
+    public NodeSourceCreationWindow(RMController controller) {
+        super(controller, "Add Node Source", "Updating available Infrastructures and Policies");
+        this.buildForm();
+    }
+
+    @Override
+    protected boolean isNodeSourceEdited() {
+        return false;
+    }
+
+    @Override
+    protected void populateFormValues(Label windowLabel, DynamicForm windowForm, TextItem nodeSourceNameItem,
+            CheckboxItem nodesRecoverableItem) {
+
+        nodesRecoverableItem.setValue(true);
+
+        controller.fetchSupportedInfrastructuresAndPolicies(() -> {
+
+            HashMap<String, List<FormItem>> allFormItemsPerPlugin = new HashMap<>();
+            LinkedHashMap<String, String> selectItemValues = new LinkedHashMap<>();
+
+            ArrayList<FormItem> allFormItems = prepareFormItems();
+
+            allFormItems.add(infrastructureSelectItem);
+            addAllPluginValuesToAllFormItems(allFormItemsPerPlugin,
+                                             allFormItems,
+                                             selectItemValues,
+                                             controller.getModel().getSupportedInfrastructures().values());
+            infrastructureSelectItem.setValueMap(selectItemValues);
+
+            allFormItems.add(new SpacerItem());
+            selectItemValues.clear();
+
+            allFormItems.add(policySelectItem);
+            addAllPluginValuesToAllFormItems(allFormItemsPerPlugin,
+                                             allFormItems,
+                                             selectItemValues,
+                                             controller.getModel().getSupportedPolicies().values());
+            policySelectItem.setValueMap(selectItemValues);
+
+            infrastructureSelectItem.addChangedHandler(changedEvent -> resetFormForInfrastructureSelectChange(allFormItemsPerPlugin));
+            policySelectItem.addChangedHandler(changedEvent -> resetFormForPolicySelectChange(allFormItemsPerPlugin));
+
+            windowForm.setFields(allFormItems.toArray(new FormItem[allFormItems.size()]));
+            windowLabel.hide();
+            windowForm.show();
+
+            hideAllFormItems(allFormItemsPerPlugin);
+
+        }, () -> window.hide());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1612)